Puzzlepart/did
# Initial status checks for PRs

### Your checklist for this pull request
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Make sure you've updated the CHANGELOG if applicable
- [x] Decide file naming patterns for our GitHub actions
- [x] Proper naming for `github/workflows/pr_install_package.yml`

### Description
Initial status checks for `dev` and `master` PRs.

This first version is very simple:

**.github/workflows/pr_install_package.yml**
```yaml
name: Status check/1
on:
  pull_request:
    types: [assigned, opened, synchronize, reopened]
    branches:
      - master
      - dev
jobs:
  build:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        node-version: [10.x]
    steps:
      - uses: actions/checkout@v1
        with:
          # Checkout the head ref instead of the PR branch that github creates.
          ref: ${{ github.head_ref }}
      - name: Use Node.js ${{ matrix.node-version }}
        uses: actions/setup-node@v1
        with:
          node-version: ${{ matrix.node-version }}
      - name: Install and build
        run: |
          npm install --production --no-progress --loglevel silent --no-shrinkwrap --no-fund   
          npm run packageClient
        env:
          NODE_ENV: production
```

When a new PR is `assigned`, `opened`, `synchronize` or `reopened` on **dev** or **master**, we'll do the following:

- Set up a new instance running on `ubuntu-latest` and `node 10.x`
- Checkout the head ref
- `npm install --production --silent`
- `npm run packageClient`

With this check we'll make sure `node_modules` installs and there's no errors in our ` client` code.

![image](https://user-images.githubusercontent.com/7606007/80699229-ee609780-8adb-11ea-802d-3d7a670d3b8a.png)

![image](https://user-images.githubusercontent.com/7606007/80700345-d68a1300-8add-11ea-8606-8052c7db6ce5.png)


### Relevant issues
#178
